### PR TITLE
📖 Add contributor ladder

### LINKED ISSUE #1529: community: Is there a contributor ladder?
Opening this to ask if there's a contributor ladder defined for this project.
Here's an example: https://github.com/kubernetes/community/blob/master/community-membership.md

Subtext: I'd be happy to help do PR review here, with the hopes of working towards maintainership, if there's a path for that :)

Previous contributions:
- https://github.com/ossf/scorecard/pull/1502
- https://github.com/ossf/scorecard/pull/1506
- https://github.com/ossf/scorecard/pull/1516

Maybe this is something like expanding CODEOWNERS, similar to https://github.com/apiclarity/apiclarity.io/pull/29.
Ideas on general org/repo mgmt (some of which you're likely already doing): https://github.com/todogroup/governance/issues/106#issuecomment-1000962064

cc: @inferno-chromium @naveensrinivasan @azeemshaikh38 @laurentsimon (ref: https://github.com/ossf/scorecard/blob/e7740151948d388e53de127d1e69a2fd616341de/.github/CODEOWNERS#L3)

### PR DESCRIPTION
**Pull request by pnacht**
#### What kind of change does this PR introduce?

Fixes #1529. Adds a contributor ladder so community members know how to increase their participation if they so desire.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

### CODE REVIEW

File: CONTRIBUTOR_LADDER.md
Comment (laurentsimon) [inline CONTRIBUTOR_LADDER.md:74]: +1 means LGTM or a comment?

File: CONTRIBUTOR_LADDER.md
Comment (laurentsimon) [inline CONTRIBUTOR_LADDER.md:169]: curious if this should be discussed prior to sending a PR

File: CONTRIBUTOR_LADDER.md
Comment (pnacht) [reply CONTRIBUTOR_LADDER.md]: Clarified. PTAL.

File: CONTRIBUTOR_LADDER.md
Comment (laurentsimon) [reply CONTRIBUTOR_LADDER.md:173]: I don't see an update about consulting with the team

Comment (pnacht) [reply CONTRIBUTOR_LADDER.md:173]: `Such requests will hopefully come after thoughtful conversations with the rest of the team and with sufficient forewarning for the others to prepare.`

Is this not what you meant?

Comment (pnacht) [reply CONTRIBUTOR_LADDER.md:173]: `Such requests will hopefully come after thoughtful conversations with the rest of the team and with sufficient forewarning for the others to prepare`. Is this not what you meant? (it doesn't appear in the diff above).

Comment (laurentsimon) [reply CONTRIBUTOR_LADDER.md:173]: It is! Sorry I did not se the update at the time I read :)

**Review by laurentsimon — APPROVED** [review]
Thanks!

@ossf/scorecard-maintainers PTAL

**Comment by spencerschrock** [comment]
> Thanks!
> 
> @ossf/scorecard-maintainers PTAL

Will take a look mid next week. Feel free to ping if I forget

**Review by di — APPROVED** [review]

**Review by naveensrinivasan — APPROVED** [review]
Thank you!

**Comment by naveensrinivasan** [comment]
> > Thanks!
> > @ossf/scorecard-maintainers PTAL
> 
> Will take a look mid next week. Feel free to ping if I forget

Let's wait for @spencerschrock 👀  before merging it.

File: CONTRIBUTOR_LADDER.md
Comment (spencerschrock) [inline CONTRIBUTOR_LADDER.md:65]: What do you think about a maintainer or 2 triagers? As written, being a community member require more sponsorship than a triager.

File: CONTRIBUTOR_LADDER.md
Comment (spencerschrock) [inline CONTRIBUTOR_LADDER.md:156]: Is there a certain time window maintainers should be given to respond? It would be a balance between giving maintainers a chance to respond (e.g. I was just on vacation for a week or two), but also not blocking if someone is too busy.

File: CONTRIBUTOR_LADDER.md
Comment (pnacht) [reply CONTRIBUTOR_LADDER.md]: True! Fixed.

File: CONTRIBUTOR_LADDER.md
Comment (pnacht) [reply CONTRIBUTOR_LADDER.md:113]: Ah, that's annoying indeed... should I remove that option?

Or maybe triagers could be added to the CODEOWNERS file in comments? They wouldn't get tagged in issues, but it'd add to transparency.

Or... is there a problem with an invalid CODEOWNERS item? Does the entire file stop working or do just those individuals not get pinged?

File: CONTRIBUTOR_LADDER.md
Comment (pnacht) [reply CONTRIBUTOR_LADDER.md:156]: Good point. I can add a minimum waiting period before the issue is closed, if you want. Two weeks?

File: CONTRIBUTOR_LADDER.md
Comment (spencerschrock) [reply CONTRIBUTOR_LADDER.md:113]: As far as I know, the rest of the `CODEOWNERS` file still works. Just when viewed through GitHub's webUI it complains that it's an invalid codeowners file with a squiggle under the user without write access.

File: CONTRIBUTOR_LADDER.md
Comment (spencerschrock) [reply CONTRIBUTOR_LADDER.md:156]: Probably best to leave as is, lest we over-specify. If it's a problem we can add it something later.

File: CONTRIBUTOR_LADDER.md
Comment (raghavkaul) [inline CONTRIBUTOR_LADDER.md:147]:
```suggestion
- Facilitating Scorecard-specific community meetings, if possible and comfortable
```

File: CONTRIBUTOR_LADDER.md
Comment (raghavkaul) [inline CONTRIBUTOR_LADDER.md:125]:
```suggestion
- adherence to API and style conventions
```

**Review by raghavkaul — APPROVED** [review]
Looks good, thank you!

**Pull request merged**
